packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -892,7 +879,22 @@ function commitUnmount(
             const {destroy, tag} = effect;
             if (destroy !== undefined) {
               if ((tag & HookPassive) !== NoHookEffect) {
-                enqueuePendingPassiveHookEffectUnmount(current, effect);
+                effect.tag |= HookHasEffect;
+
+                current.effectTag |= Passive;
+
+                if (__DEV__) {
+                  // This flag is used to avoid warning about an update to an unmounted component
+                  // if the component has a passive unmount scheduled.
+                  // Presumably the listener would be cleaned up by that unmount.
+                  current.effectTag |= PassiveUnmountPendingDev;
+                  const alternate = current.alternate;
+                  if (alternate !== null) {
+                    alternate.effectTag |= PassiveUnmountPendingDev;
+                  }
+                }
+
+                schedulePassiveEffectCallback();
```

Comment:
Don't think we need this now that we check and schedule at the root, too.

Reply:
It's still necessary.

Reply:
Could you explain why? I believe you, just curious.

The reason I guess this would work is because a Deletion tag results in a PassiveSubtree tag on all the ancestors up to the root, which we would then detect with this check: https://github.com/facebook/react/pull/19374/files#diff-71727dd0c0ae3972b0f7725a87294f4cR2186

(Not a blocker regardless, btw)

Reply:
Yeah. My bad, I should have left a more thorough response initially.

A `Deletion` effect would get bubbled into a `PassiveSubtree` tag, but that bubbling happens during the render phase, whereas this function is in the commit phase. (There's no bubbling after this point.)

Reply:
There's a Deletion effect on the ancestor that got deleted, though, which does get bubbled during render phase

Reply:
The `return` pointer has been cleared already though

Reply:
Sorry, please disregard my other comment. I thought I was responding to a different thread when I got the email notification. :smile: